Add wildcard (regex) support to TestNames parameter

andyleejordan opened the pull request:
I still need to test this, but it seemed simple enough to enable wildcard support for `Testnames`.

chvalean commented on Testscripts/Linux/xfstesting.sh:
```diff
@@ -0,0 +1,161 @@
+    if [ -e ${XFSTestConfigFile} ]; then
+        LogMsg "${XFSTestConfigFile} file is present."
+    else
+        LogErr "Error: missing ${XFSTestConfigFile} file"
```
remove error:

chvalean commented on Testscripts/Windows/FILE-SYSTEM-VERIFICATION-TESTS.ps1:
```diff
@@ -1,15 +1,33 @@
+# Description:
+#     This Powershell script will run xfstexting.sh bash script
+#     - It will construct the config file needed by xfstesting.sh
+#     - It will start xfstesting.sh. Max allowd run time is 3 hours.
```
allowed*

chvalean commented on Testscripts/Linux/xfstesting.sh:
```diff
@@ -0,0 +1,161 @@
+            pack_list="libacl1-dev libaio-dev libattr1-dev libgdbm-dev libtool-bin libuuid1 libuuidm-ocaml-dev sqlite3 uuid-dev uuid-runtime xfslibs-dev zlib1g-dev"
+        ;;
+
+        redhat*|centos*)
```
also add fedora here

chvalean commented on Testscripts/Windows/FILE-SYSTEM-VERIFICATION-TESTS.ps1:
```diff
@@ -1,15 +1,33 @@
-param([object] $AllVmData, [object] $CurrentTestData)
+#
+# Description:
+#     This Powershell script will run xfstexting.sh bash script
```
testing*

chvalean commented on Testscripts/Windows/FILE-SYSTEM-VERIFICATION-TESTS.ps1:
```diff
@@ -22,37 +40,59 @@ function Main {
-        if ($XFSTestingConsole -imatch "Passed all") {
-            $testResult = "PASS"
+        # Get logs. An extra check for the previous $state is needed
+        # The test could actually be hanged. If state.txt is showing
```
could actually hang*

adriansuhov commented:
**NVMe Ubuntu 16.04 results**
[LISAv2 Test Results Summary]
Test Run On           : 02/04/2019 10:25:03
ARM Image Under Test  : Canonical : UbuntuServer : 16.04-LTS : latest
Total Test Cases      : 4 (4 Pass, 0 Fail, 0 Abort)
Total Time (dd:hh:mm) : 0:1:50

   ID TestCaseName                                                                TestResult TestDuration(in minutes)
---------------------------------------------------------------------------------------------------------------------
    1 NVME-FILE-SYSTEM-VERIFICATION-GENERIC                                             PASS                57.32
    2 NVME-FILE-SYSTEM-VERIFICATION-XFS                                                 PASS                27.57
    3 NVME-FILE-SYSTEM-VERIFICATION-EXT4                                                PASS                 7.22
    4 NVME-FILE-SYSTEM-VERIFICATION-BTRFS                                               PASS                10.62

andyleejordan committed:
Add regex expression support to -TestNames parameter

However, it also has logic so as to not match substrings without an
explicit pattern to do so.

LiliDeng reviewed (dismissed):
Better to document this usage, I test VERIFY-DEPLOYMENT-PROVISION and VERIFY-DEPLOYMENT-PROVISION.*, both work.

andyleejordan committed:
Clean up Markdown syntax in README

This cleans up the lists and indentation so that the README renders
appropriately on GitHub.

andyleejordan committed:
Add example for -TestNames usage with regular expression matching
